src/govuk/components/notification-banner/notification-banner.yaml

```diff
@@ -0,0 +1,177 @@
+params:
+- name: text
+  type: string
+  required: true
+  description: If `html` is set, this is not required. Text to use within the notification banner. If `html` is provided, the `text` argument will be ignored.
+- name: html
+  type: string
+  required: true
+  description: If `text` is set, this is not required. HTML to use within the notification banner. If `html` is provided, the `text` argument will be ignored.
+- name: title
+  type: string
+  required: false
+  description: Title text to use within the notification banner. Defaults to 'Important' ('Success' for success type and 'Error' for error type). If `titleHtml` is supplied, the `title` argument will be ignored.
+- name: titleHtml
+  type: string
+  required: false
+  description: Title HTML to use within the notification banner. If `titleHtml` is provided, the `title` argument will be ignored.
+- name: titleHeadingLevel
+  type: string
+  required: false
+  description: Heading level, from 1 to 6. Default is `2`.
+- name: type
+  type: string
+  required: false
+  description: If `type` is set to `success` or `error`, the notification banner sets `role` to `alert` and `tabindex` to `-1`, and JavaScript moves the keyboard focus moves to the notification banner when the page loads. If `type` is not set, the notification banner defaults to setting `role` to `region` and using `aria-labelledby` to provide information for users of assistive technologies.
+- name: role
+  type: string
+  required: false
+  description: Overrides the value of the `role` attribute for the notification banner. Defaults to `region`. If `type` is set to `success` or `error`, defaults to `alert`.
+- name: tabindex
+  type: string/boolean
+  required: false
+  description: Overrides the value of the `tabindex` attribute for the notification banner. Not set by default.  If `type` is set to `success` or `error`, defaults to `-1`; the attribute can be removed by setting `tabindex` to `false`;
+- name: titleId
+  type: string
+  required: false
+  description: Overrides the value of the `id` attribute for the title. `id` used by the `aria-labelledby` attribute on the notification banner to provide information to users of assistive technologies. `id` defaults to `govuk-notification-banner-title` if `role` is set to `region`. If `type` is set to `success` or `error`, `id` is not rendered by default.
```

Comment:
Could break up the second sentence. So: “...on the notification banner. This provides information…”

Should this display first? “id used by the aria-labelledby attribute on the notification banner.” It defines the option, which might be the first info a user wants to know.

Replace passive? So, “if you set role to region” instead of “if role is set to region” and “If you set type…" instead of "If type is set..."